<launch>
  
  <arg name="nao_ip" default="$(optenv NAO_IP 192.168.1.127)"/>
  <arg name="nao_port" default="$(optenv NAO_PORT 9559)"/>
  
  <!--
    Top camera 1
  -->
  <node pkg="nao_driver" type="nao_camera.py" name="nao_camera1" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <remap from="image_raw" to="image_raw1"/>
  </node>
  
  <!--
    IMAGE CONVERTER 1
  -->
  <node pkg="nao_behavior_tree" type="imageconverter" name="im_conv1" args='1' output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <!--
    Sonar 1
  -->
  <node pkg="nao_behavior_tree" type="nao_sonar" name="nao_sonar1" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
    <remap from="sonar" to="sonar1"/>
  </node>
  
  <!--
    Leds 1
  -->
  <node pkg="nao_behavior_tree" type="nao_leds" name="nao_leds1" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <!--
    Walker 1
  -->
  <node pkg="nao_driver" type="nao_walker.py" name="nao_walker1" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" output="screen">
  	<remap from="cmd_vel" to="cmd_vel1"/>
  </node>
  
  
  <!--
    Actions 1
  -->
  <node pkg="nao_behavior_tree" type="Greet" name="Greet1" args="1">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="Search" name="Search1" args="1" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="GoClose" name="GoClose1" args="1" output="screen">
    <rosparam command="load" file="$(find nao_behavior_tree)/launch/parameters.yaml"/>
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="Stop" name="Stop1" args="1">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="StopWalking" name="StopWalking1" args="1">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>
  
  <node pkg="nao_behavior_tree" type="TakeBall" name="TakeBall1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>


  <node pkg="nao_behavior_tree" type="Catch" name="Catch1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>

  <node pkg="nao_behavior_tree" type="GiveBall" name="GiveBall1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>

  <node pkg="nao_behavior_tree" type="Drop" name="Drop1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip)"/>
    <param name="NAO_PORT" value="$(arg nao_port)"/>
  </node>

</launch>
